src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -356,52 +356,56 @@ protected void CheckDocumentIsOpen(DocumentId documentId)
         }
 
         protected internal void OnDocumentOpened(
-            DocumentId documentId, SourceTextContainer textContainer,
-            bool isCurrentContext = true)
+            DocumentId documentId, SourceTextContainer textContainer, bool isCurrentContext = true)
         {
-            using (_serializationLock.DisposableWait())
-            {
-                CheckDocumentIsInCurrentSolution(documentId);
-                CheckDocumentIsClosed(documentId);
+            SetCurrentSolution(
+                static (oldSolution, data) =>
```

Comment:
in many cases i made the large callbacks static. this was to ensure that it was very clear how they were using 'this' and the argumetns passed in.  making them static meant we had to be very explicit about what we were calling and when which helped me with reasoning about things.